### PR TITLE
lib: nrf_modem: don't enforce socket priority over native TLS

### PR DESCRIPTION
**Pull request by mniestroj**
Commit 27e7ebd84722 ("lib: nrf_modem: Allow to create native sockets
with offloading enabled") introduced a build time check to enforce
offloaded socket priority over Zephyr native TLS.

In some cases native TLS might be the preferred choice, so remove that
enforcement and allow application developer to decide whether offloaded
TLS or native TLS should be used by default.

### CODE REVIEW

**Comment by mniestroj** [comment]
Related Devzone issue: https://devzone.nordicsemi.com/f/nordic-q-a/84041/bug-changes-to-nrf91_sockets-c-breaking-native-tls-when-using-libraries.

This PR will allow to use native TLS on application configuration level, instead of having to pass custom flags usign setsockopt() calls.

**Review by rlubos — APPROVED** [review]
Swapping the priorities means that the app will no longer be able to create an offloaded TLS socket (until we get the dispatcher in), but I guess you're right that we should not prohibit this.

**Comment by mniestroj** [comment]
Is [CI/Jenkins/NRF](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-7325/2/) blocking this PR from merging? Or is it internal review process that we are waiting for? Thanks!

**Comment by rlubos** [comment]
Just waiting for the second approval, @lemrey would you mind taking a look?

@mniestroj In a meantime, would be good if you could rebase the PR, we had some changes in our CI recently and the GH statuses are not updated properly here.

**Comment by lemrey** [comment]
Should the socket priority for nRF91 be a Kconfig then?

**Review by lemrey — APPROVED** [review]

**Pull request merged**
